react-native/tslint.json

```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "../tslint.json",
+    "rules": {
+        // This package uses the Function type for good reason and `object` is TypeScript 2.2 only.
+        "forbidden-types": false
```

Comment:
In that case, the output should have an identical type as the input. So you can take a `<T extends Function>` and return a `T`.